File: website/src/main/java/pl/wilenskid/api/model/bean/SubPostBean.java
package pl.wilenskid.api.model.bean;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import lombok.Data;

import java.util.List;

@Data
@JsonAutoDetect
public class SubPostBean {
  private Long id;
  private Long postId;
  private String title;
  private FileBean description;
  private FileBean content;
  private String created;
  private String updated;
}

File: website/src/main/typescript/app/front-end/api/models/sub-post-bean.ts
import { UserBean } from './user-bean';
import { FileBean } from './file-bean';

export interface SubPostBean {
	readonly id: number;
	readonly postId: number;
	readonly title: string;
	readonly description: string;
	readonly content: FileBean;
	readonly contributors: UserBean[];
	readonly created: string;
	readonly updated: string;
}
